airflow-core/src/airflow/models/deadline.py

```diff
@@ -143,7 +137,7 @@ class BaseDeadlineReference(LoggingMixin, ABC):
         def reference_name(cls: Any) -> str:
             return cls.__name__
 
-        def evaluate_with(self, **kwargs: Any) -> datetime:
+        def evaluate_with(self, session: Session, interval: timedelta, **kwargs: Any) -> datetime:
```

Comment:
```suggestion
        def evaluate_with(self, *, session: Session, interval: timedelta, **kwargs: Any) -> datetime:
```

This applies to all `evaluate_with` and `_evaluate_with`. Make these all keyword-only arguments since they are only used as such, and this avoids potential future headaches.